### DISCUSSION TITLE
A RecursionError (maximum recursion depth exceeded) occurred.

### DISCUSSION THREAD

**Discussion by Victor94-king** [pyinstaller/pyinstaller#7554] [Help] [answered]
Hi,

I faced a problem of "A RecursionError (maximum recursion depth exceeded) occurred." and I have tried solution like changing my.spec file and running PyInstaller with the .spec file  u have suggested and it did not work.
![image](https://user-images.githubusercontent.com/103173758/230766386-fde849fe-bd57-4098-95e5-33028fbf94db.png)

My code is shown [github](https://github.com/Victor94-king/ComputerVersion/blob/main/main.py) , could you plz help me find it out?

**Comment by bwoodsend** [comment, marked as answer]
I suggest you use a minimal virtual environment with only the packages you need installed.

**Reply by Victor94-king** [reply]
thanks for u reply , and do u mean that my environment contains too much packages?

**Reply by bwoodsend** [reply]
*Yes* is the short answer.

Pandas contains tonnes of optional dependencies, including matplotlib which supports a wide range of GUI backends including Jupyter notebooks which depends on various webserver libraries and code completion engines and its own widgets libraries. This forms a tree of dependencies which usually ends up including your entire Python environment. PyInstaller has no way of knowing which parts of that tree of optional dependencies you really want so it'll include whatever it can find. If the tree it too deep, it'll eventually hit the recursion limit. If you take away all the packages you don't want, then PyInstaller can't navigate itself so far down the dependency tree that it hits the recursion limit.
